### PR TITLE
Mega huge (annoying) pr, sorry guys

### PR DESCRIPTION
**Pull request by chrisnojima**
cry

@keybase/react-hackers 

Ok so this big pr does a couple of things:

- Changes how parseRoute works. You only get the currentPath and the uri, you don't give a mapStateToProps anymore as this was causing issues with react thrashing and remaking components over and over. Instead we export connected components and use that for what we were doing in parseRoute (setup the store, hook actions into dispatch). This works with dynamic stores as well
- Changed the nav slightly. Desktop and ios will only render the foremost tab and not all tabs
And since I had to literally touch most every file we have I did

- Remove direct react-native requires (yay desktop)
- Remove extra whitespace in things like { Component, View } from 'base-react'

Do yourself a favor and use the magical w=1 flag to ignore all the whitespace changes
https://github.com/keybase/client/pull/1350/files?w=1
(although github still won't let you comment in this mode...)

### CODE REVIEW

File: react-native/react/actions/login2.js [hunk 1]
```diff
@@ -219,6 +219,7 @@ function askForPaperKey (cb) {
 // Show a device naming page, call cb() when done
 // existing devices are blacklisted
 function askForDeviceName (existingDevices, cb) {
+  // NOJIMA TODO WHAT IS GOIN ON HERE? are 2 calls happening, is one dead? AAAAAA<<<<<<<<<<<<<<<<
```
Comment (chrisnojima) [inline react-native/react/actions/login2.js]: Yeah i have to figure this out, will do 2moro

- [ ] fix this

File: react-native/react/login2/register/set-public-name/index.mobile.js [hunk 2]
```diff
@@ -67,3 +58,30 @@ SetPublicName.propTypes = {
...
+    const {deviceName} = state.login2
+    return {deviceName}
+  },
+  dispatch => {
+    return {onSubmit: deviceName => dispatch(setDeviceName(deviceName))}
+  }
+)(SetPublicName)
+*/
+
+// NOJIMA TODO this isn't consistent
```
Comment (chrisnojima) [inline react-native/react/login2/register/set-public-name/index.mobile.js]: - [ ] fix this

File: react-native/react/router/meta-navigator.js [hunk 3]
```diff
@@ -5,18 +5,29 @@
...
 import Immutable from 'immutable'
+import {connect} from '../base-redux'
 
 class MetaNavigator extends Component {
-  constructor () {
-    super()
+  constructor (props) {
+    super(props)
 
+    console.log('NOJ meta-nav will construct')
```
Comment (chrisnojima) [inline react-native/react/router/meta-navigator.js]: - [ ] remove this stuff from this file. just to show when things are made / destroyed. now it happens when we expect!

**Comment by cjb** [comment]
LGTM!

File: react-native/react/actions/config.mobile.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
 'use strict'
 
-import { NativeModules } from 'react-native'
+import {NativeModules} from '../base-react'
 import * as Constants from '../constants/config'
-import { autoLogin } from './login2'
+import {autoLogin} from './login2'
```
Comment (MarcoPolo) [inline react-native/react/actions/config.mobile.js]: a one point item for next sprint should be to enforce this with the linter.

File: react-native/react/debug/index.ios.js [hunk 5]
```diff
@@ -1,42 +0,0 @@
...
-    return (
-      <Temp navigator={this.props.navigator} {...this.state.mock} />
-    )
-    // return React.createElement(require('./briding-tabs'))
-  }
-}
-
-Debug.propTypes = {
-  navigator: React.PropTypes.object
-}
```
Comment (MarcoPolo) [inline react-native/react/debug/index.ios.js]: :red_circle: :red_circle: :fire: :smile:

File: react-native/react/login2/register/set-public-name/index.mobile.js (hunk 2 above)
Comment (chrisnojima) [inline react-native/react/login2/register/set-public-name/index.mobile.js]: actually fixing this in the kex fixes after this pr

File: react-native/react/actions/config.mobile.js (hunk 4 above)
Comment (chrisnojima) [inline react-native/react/actions/config.mobile.js]: https://keybase.atlassian.net/browse/DESKTOP-81

**Comment by MarcoPolo** [comment]
:recycle: :factory: 
:+1:

**Pull request merged**